File: matrix.exponential/R/matrix.exponential.v4.R

matexp <- function(X,m)
{
  n <- dim(X)[1]
  Z <- diag(n)
  Y <- Z
  for(k in 2:m)
  {
      Z <- Z%*%X/(k-1)
      Y <- Y + Z
  }
return(Y)
}

args <- commandArgs(trailingOnly=TRUE)

n <- as.numeric(args[1])
m <- as.numeric(args[2])
s <- as.numeric(args[3])
stem <- args[4]

set.seed(s)

X <- matrix(runif(n^2),n,n)
Y <- matexp(X,m)


fileX <- paste(stem,".X.n=",n,".m=",m,".s=",s,".RData",sep="")
fileY <- paste(stem,".Y.n=",n,".m=",m,".s=",s,".RData",sep="")


save(X,file=fileX)
save(Y,file=fileY)

File: matrix.exponential/python/matrix.exponential.v4.py
import numpy as np
import sys


def matexp(X,m) :
    n,_ = np.shape(X)
    Z = np.identity(n)
    Y = Z
    for k in range(1,m) :
        Z = np.matmul(Z,X)/k
        Y = np.add(Y,Z)
    return Y
    

n = int(sys.argv[1])
m = int(sys.argv[2])
s = int(sys.argv[3])
stem = str(sys.argv[4])

np.random.seed(s)

X = np.random.rand(n,n)
Y = matexp(X,m)


fileX = stem + ".X.n=" + str(n) + ".m=" + str(m) + ".s=" + str(s)
fileY = stem + ".Y.n=" + str(n) + ".m=" + str(m) + ".s=" + str(s)


np.save(fileX,X)
np.save(fileY,Y)







